### ISSUE TITLE
Robot name: add test to check names are generated randomly

### ISSUE THREAD

**Issue by rchavarria** [exercism/xecmascript#177]
`README` file says that the solution should generate robot names randomly, but it's not forced by any test. Add such a test that check, at least, that robot names are not generated sequentially.

**Comment by matthewmorgan** [comment]
Just wondering how we can test this?  I think it may not be possible. 

It's entirely possible (though very unlikely) that a set of randomly generated names could in fact be sequential.  How can we tell the difference?

imagine putting 10 names in a hat and drawing them 'randomly'.  It is certainly conceivable that the names could be drawn in alphabetical order.

**Comment by rchavarria** [comment]
Absolutely. I think that discussion already happened on xjavascript track (well kind of [xjavascript#105](https://github.com/exercism/xjavascript/issues/105))

Let's assume names are generated randomly. Testing two robot names has a certain probability to be sequential: `AA123` and `AA124`. But, testing three names, the probability will be much lower: `AA123`, `AA124` and `AA125` is much more difficult to get than the previous one.

I'm not sure how to mathematically calculate each probability. 

And it might not be so simple to test that two names are sequential. It's easy with `AA123` and `AA124`, but it's not so easy with `AA999` and `AB000`. It will make the test even more difficult than the solution itself.

**Comment by matthewmorgan** [comment]
Yeah, I guess it's just a little weird-- I could have a functioning solution that generates a random sequence, and it could fail bc my solution generates two (or three) names that just happen to be sequential.

BTW I think the chance of three successive sequential names is (1/675999)*(1/675998), or 2.19 x 10^-12.  About 2 in a trillion.  Really small.

**Comment by rchavarria** [comment]
Agree. The probability to have 3 succesive sequential names is really small. It could work.

But, the problem is how to check that two given names are sequential. I can only think of a solution that almost solves the exercise itself, and that should not be the goal of tests.

**Comment by matthewmorgan** [comment]
How about something like this?  Obviously the test isn't set up to test three names in this example, but this is the principle:

```
  it('should test for sequential names', () => {
    expect(areSequential("AA000", "AA001")).toEqual(true);
    expect(areSequential("AA000", "AA002")).toEqual(false);
    expect(areSequential("AA999", "AB000")).toEqual(true);
    expect(areSequential("AA999", "AA998")).toEqual(true);
    expect(areSequential("AA999", "AA997")).toEqual(false);
    expect(areSequential("AB000", "AA999")).toEqual(true);
    expect(areSequential("AB999", "AA000")).toEqual(false);
  })


  const areSequential = (name1, name2) => {
    const alpha1 = name1.substr(0,2);
    const alpha2 = name2.substr(0,2);
    const num1 = +name1.substr(2,3);
    const num2 = +name2.substr(2,3);

    const numDiff = num2-num1;
    const alphaDiff = (alpha2.charCodeAt(0)-alpha1.charCodeAt(0))*26
        + (alpha2.charCodeAt(1)-alpha1.charCodeAt(1));

    const totalDiff = alphaDiff*1000 + numDiff;

    return Math.abs(totalDiff) <= 1;
  };
```

**Comment by rchavarria** [comment]
Yeah! Exactly. `areSequential` is a method we could use in the test to check that two or three robot names are not consecutive.

In fact, I've been thinking, and I came up with a solution quite similar to your `areSequential` (but I have no code yet, you win). The idea is to calculate a numerical decimal value of the given name. That value would be calculated as if it was a natural number. Given a robot name `BC123`:

- `123` matches the decimal value, it is `123`
- The least valuable character, `C`, can be assigned a decimal value o `1000`. So a `A` would add `0`, a `B` would add `1000` and a `C` would add `2000`.
- Most valuable characters would be valued as `26 * 1000`. `A` is a `0`, `B` is a `26000` and a `C` is a `52000`.

Adding all those values, `BC123` would be represented as `26000 + 2000 + 123 = 28123`. Once the decimal value has been calculated, it's easy to know if they are consecutive or not. As your `areSequential` does. Well done!

I think we have all pieces together. Would you mind to add the test and create a PR?

**Comment by matthewmorgan** [comment]
it sounds like our ideas are very similar, so great.

I am happy to create the tests and PR.

**Comment by matthewmorgan** [comment]
Closed by #180.
